pkg/domain/infra/abi/images.go

```diff
@@ -806,3 +817,32 @@ func localPathFromURI(url *url.URL) (string, error) {
 	}
 	return url.Path, nil
 }
+
+func createSignature(m []byte, dockerReference string, mech signature.SigningMechanism, keyIdentity string) ([]byte, error) {
+	newSig, err := signature.SignDockerManifest(m, dockerReference, mech, keyIdentity)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating new signature")
+	}
+	return newSig, nil
+}
+
+func putSignature(sigStoreDir string, signature []byte, instanceDigest digest.Digest) error {
+	signatureDir := fmt.Sprintf("%s@%s=%s", sigStoreDir, instanceDigest.Algorithm(), instanceDigest.Hex())
+	if err := os.MkdirAll(signatureDir, 0751); err != nil {
+		// The directory is allowed to exist
+		if !os.IsExist(err) {
+			logrus.Errorf("error creating directory %s: %s", signatureDir, err)
+			return err
```

Comment:
The caller is already calling `logrus.Errorf` on this error. Generally I would expect such a decision of whether to return or log the error to be maintained at a single level of the call stack.

(Same for the other errors.)